Tasks/DotNetCoreCLIV2/dotnetcore.ts

```diff
@@ -325,22 +326,50 @@ export class dotNetExe {
         }
 
         var projectFiles = utility.getProjectFiles(projectPattern);
+        var resolvedProjectFiles: string[] = [];
 
         if (searchWebProjects) {
-            projectFiles = projectFiles.filter(function (file, index, files): boolean {
+            resolvedProjectFiles = projectFiles.filter(function (file, index, files): boolean {
                 var directory = path.dirname(file);
                 return tl.exist(path.join(directory, "web.config"))
                     || tl.exist(path.join(directory, "wwwroot"));
             });
 
-            if (!projectFiles.length) {
-                tl.error(tl.loc("noWebProjectFound"));
-            }
+            if (!resolvedProjectFiles.length) {
+                var projectFilesUsingWebSdk = projectFiles.filter(this.isWebSdkUsed)
+                if(!projectFilesUsingWebSdk.length) {
+                    tl.error(tl.loc("noWebProjectFound"));
+                }
+                return projectFilesUsingWebSdk;
+            } 
+            return resolvedProjectFiles;
         }
-
         return projectFiles;
     }
 
+    private isWebSdkUsed(projectfile: string): boolean {
+        try {
+            var fileBuffer: Buffer = fs.readFileSync(projectfile);
+            var webConfigContent: string;
+            
+            var fileEncodings = ['utf8', 'utf16le']
+
+            for(var i = 0; i < fileEncodings.length; i++) {
+                tl.debug("Trying to decode with " + fileEncodings[i])
+                webConfigContent = fileBuffer.toString(fileEncodings[i])
+                try {
+                    var projectSdkUsed: string = ltx.parse(webConfigContent).getAttr("sdk") || ltx.parse(webConfigContent).getAttr("Sdk")
```

Comment:
you can consider ending the lines with `;`. and to have help, you can add ts-lint in vs code as an extension.

Reply:
That extension won't let the work get done 🐐 
But yeah, will endlines with semicolon